analyzers/rspec/cs/S5443_c#.html

```diff
@@ -0,0 +1,71 @@
+<p>Operating systems have global directories where any user has write access. Those folders are mostly used as temporary storage areas like
+<code>/tmp</code> in Linux based systems. An application manipulating files from these folders is exposed to race conditions on filenames: a malicious
+user can try to create a file with a predictable name before the application does. A successful attack can result in other files being accessed,
+modified, corrupted or deleted. This risk is even higher if the application runs with elevated permissions.</p>
+<p>In the past, it has led to the following vulnerabilities:</p>
+<p> * <a href="https://nvd.nist.gov/vuln/detail/CVE-2012-2451">CVE-2012-2451</a></p>
+<p> * <a href="https://nvd.nist.gov/vuln/detail/CVE-2015-1838">CVE-2015-1838</a></p>
+<p>This rule raises an issue whenever it detects a hard-coded path to a publicly writable directory like <code>/tmp</code> (see examples bellow). It
+also detects access to environment variables that point to publicly writable directories, e.g., <code>TMP</code>, <code>TMPDIR</code> and
+<code>TEMP</code>.</p>
+<p> * <code>/tmp</code></p>
+<p> * <code>/var/tmp</code></p>
+<p> * <code>/usr/tmp</code></p>
+<p> * <code>/dev/shm</code></p>
+<p> * <code>/dev/mqueue</code></p>
+<p> * <code>/run/lock</code></p>
+<p> * <code>/var/run/lock</code></p>
+<p> * <code>/Library/Caches</code></p>
+<p> * <code>/Users/Shared</code></p>
+<p> * <code>/private/tmp</code></p>
+<p> * <code>/private/var/tmp</code></p>
+<p> * <code>\Windows\Temp</code></p>
+<p> * <code>\Temp</code></p>
+<p> * <code>\TMP</code></p>
+<p> * <code>%USERPROFILE%\AppData\Local\Temp</code></p>
+<h2>Ask Yourself Whether</h2>
+<ul>
+  <li> Files are read from or written into a publicly writable folder </li>
+  <li> The application creates files with predictable names into a publicly writable folder </li>
+</ul>
+<p>There is a risk if you answered yes to any of those questions.</p>
+<h2>Recommended Secure Coding Practices</h2>
+<p>Out of the box, .NET is missing secure-by-design APIs to create temporary files. To overcome this, one of the following options can be used:</p>
+<p> * Use a dedicated sub-folder with tightly controlled permissions</p>
+<p> * Created temporary files in a publicly writable folder and make sure:</p>
+<p> <strong></strong> Generated filename is unpredictable</p>
+<p> <strong></strong> File is readable and writable only by the creating user ID</p>
+<p> <strong></strong> File descriptor is not inherited by child processes</p>
+<p> <strong></strong> File is destroyed as soon as it is closed</p>
+<h2>Sensitive Code Example</h2>
+<pre>
+using (var writer = new StreamWriter("/tmp/f")) // Sensitive
+{
+// ...
+}
```

Comment:
_Optional:_ New syntax to reduce parenthesis hell in the demonstration?
```suggestion
using var writer = new StreamWriter("/tmp/f"); // Sensitive
```